dm/syncer/syncer.go

```diff
@@ -3572,10 +3572,42 @@ func (s *Syncer) Resume(ctx context.Context, pr chan pb.ProcessResult) {
 	s.Process(ctx, pr)
 }
 
+// CheckCanUpdateCfg check if task config can be updated.
+// 1. task must not in a pessimistic ddl state.
+// 2. only balist, route/filter rules and syncerConfig can be update at this moment.
+func (s *Syncer) CheckCanUpdateCfg(newCfg *config.SubTaskConfig) error {
+	s.RLock()
+	defer s.RUnlock()
+	// can't update when in sharding merge
```

Comment:
in some optimistic case, like partially drop column, BAlist should also not be updated

@GMHDBJD PTAL, can you list the forbidden cases?

Reply:
i think it's hard to list or check all the case ... maybe let the user to bear the risk of changing config

Reply:
also remember to add this limitation to doc